vulcanai/utils.py

```diff
@@ -419,3 +430,163 @@ def stitch_datasets(df_list, on, index_list=None):
     merged_df.fillna(0, inplace=True)
 
     return merged_df
+
+
+def feature_directionality(features, data, val_x, val_y, dense_net, filename):
+    """
+    Will conduct tests to figure out directionality of features by finding all the unique feature values present in the
+    dataset and setting the feature for all examples to every unique value of that feature.
+
+    Args:
+        features: list of all features in the model
+        data: dataframe with the data, features, y values
+        val_x: the features of the examples in the validation set
+        vay_y: the labels of the examples in the validation set
+        filename: the name of the file to save the test results to
+    """
+    test_df = None
+    for j in range(len(features)):
```

Comment:
I think this can be sped up with just looking at the train and val features as a collective without the split inner loop (so we get from O(n^2) > O(n))

Reply:
I have to do one test run for each unique value of every feature while all other feature values remain unchanged. I'm not able to see how it can be done with less complexity that what I have. Can you explain further?

Reply:
oh I see sorry I misunderstood. in that case, we should be wary of features that have a lot of values so we can stratify it to not slow down the whole process too much.